core/templates/services/math-interactions.service.ts

```diff
@@ -524,6 +524,42 @@ export class MathInteractionsService {
   // The input terms to this function should be the terms split by '+'/'-'
   // from an expression.
   termsMatch(term1: string, term2: string): boolean {
+    let getConstants = function(expression: string): string[] {
+      let constants = expression.match(/(\d*\.\d+)|(\d+)/g);
+      if (constants !== null) {
+        return constants;
+      }
+      return [];
+    };
+
+    let partsMatch = function(part1: string, part2: string): boolean {
+      // Nerdamer simplifies constant terms automatically so to avoid that,
+      // we'll replace all constants with a variable.
+      let usedVariables = nerdamer(part1).variables().concat(
```

Comment:
Instead of doing the bookkeeping, would it be more error-proof to just use a set of variables for the replacements (e.g. unicode chars) that aren't part of the global set of allowed variables? (If so, add a test to ensure that the two lists don't overlap.)

Reply:
That's what I was going to do initially but we can't use unicode chars other than alphabets since nerdamer will throw an error for them.
I've now replaced this approach with a simpler one where I just append a dummy string in front of the constants effectively making them valid variables. `3` => `const3`. `3.4` => `const3point4`.

There is one caveat to this approach though (and actually with the previous one as well) -- terms like `3 * 10^(-5)` won't match with `3 / 10^5`. This is because the 2nd part of these terms would be `const10^(-const5)` and `(const10^const5)^(-const1)` and so the -1 won't get simplified by being multiplied with 5 and the exponents won't match.

But, I believe this is the expected behavior as well, so it should be good, right @seanlip?

Reply:
Yup I think this is fine for now. Is this added as a test case?

Reply:
Yes.